### PR TITLE
Convert TextFieldWithLabelFormGroup.test.tsx to RTL

### LINKED ISSUE #28: Convert shared/components/input/TextFieldWithLabelFormGroup.test.tsx to RTL

### PR DESCRIPTION
**Pull request by mpeyper**
Fixes #28

### CODE REVIEW

File: src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx [hunk 1]
```diff
@@ -18,101 +19,89 @@ describe('text field with label form group', () => {
...
-          isRequired={expectedRequired}
+          isRequired
           onChange={jest.fn()}
         />,
       )
 
-      const label = wrapper.find(Label)
-      expect(label).toHaveLength(1)
-      expect(label.prop('isRequired')).toBeTruthy()
+      expect(screen.getByTitle(/required/i)).toBeInTheDocument()
```
Comment (mpeyper) [inline src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx:37]: Not sure if this was the best way to test if the field was required, but without the `prop` I struggled to find alternatives.

[Here is the playground](https://testing-playground.com/#markup=DwEwlgbgfKkAQGMA2BDAzmgvAIgGYHsAnAWwFoBzQ-AVwAdsZxphUAjAUyUVQxwJNJtO2OPxwAXdmnEAVdgA9xAMTCcQI8WHFJ2OGQAswaOEbgo4hdgEdqYSyBMA7WtXENJ04GDjSAnjpwEfCQiAC4LdhAAbgZgNAhyM0IwFFJDEBB2RwlCanYRAgRqNBRWALwUJDR8uBAUcVTaS1wweT50ETqG0jAg7Ox0SWS0AGsRZHQsbHjyHsckMEd2UlJcczXSQfZhkdFUgHdSAEYANjgRKnKwYnIReWIkRyn9cXFaUIB6D-2fgDp9gDMvyI5A+ACYAAxQj4zEQQVT7ABC+Da2AhcHRAFYjmC4NiwSI-OV9OwwOQXuEAOy0eRROAQbaaBCVTYLciOcLifC0OnEFCEciLQTsXDicJgmkxGC0er6URgJBIQLUQiWRziADCwSInRwAFkACyUgAcvzBRzgAIBBt+EIAnACADJWs5gzEnJBHA1gs1HUgm20OhBHI6-Sl2zGkE6-I2UuBHTG-O0nP3m34A40GuB2s0AsGkK22gGUpDHHMG40FoGUg1HBBRsNgysh34nTEA0hpyn5hMxu12qtmsGU0imzFgp1g5Px43RylHY1IIFGyOnH2Qg0aqc5yknAHx0MQlNwJumh1x9HDxMVjH6Atnk71o7L90Yzs2pv7ltH1NA-tZsEbWxA1HXNM4FznBc4EpSkzTbeMIR9E4IUpJ9Q0xI8GztXdO2jA1+wLX5n0HMFhzgMcJ1LI5y2NS1q1res5ybA9W3bTtfndXEwWjV8cwnfdCwhYtHSOXcT1fQtwzom17QBNCwwjOA8Jg45E2TC00wzA1SD4vM6KLEtqJjWjJIYpjaJbNt9y7c142vft9NIuMKMnU5LTdM1jTBUtl0pTF4xOdcIU3c112LPEoSTAE7RPCdbROLN8RPAEeNxfF9AzJMU2DF8zkvD9jQ7b9jzi-tI0AjivVA6c80TJtvOfU1KQBfzMyIkNg3Qo8lKTMTuL7GKgWfRzh1HDjKKMitBxrOtzOOUMrJPDi91wjjMQHXT80ElKAC9Yg+GVxH0GAYQSE6wBOoQkBOphGHgCZeDwIgyEoGh6BgSRFH5dhzCofYpgNcYeCmfgyD6cRLn2z6GksFA7ogbhJhwRYIEqMAQFWdhIlYFAEDGG7IAJ5gPlu4AScJoA) if anyone wants to make a suggestion?

File: src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx [hunk 2]
```diff
@@ -18,101 +19,89 @@ describe('text field with label form group', () => {
...
-          name={expectedName}
+          name="test"
           label="test"
-          value={expectedValue}
-          isEditable={false}
-          onChange={handler}
+          value=""
+          isEditable
+          onChange={(e) => {
+            callBackValue += e.target.value
```
Comment (mpeyper) [inline src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx:97]: I'm not sure what was going on here, but the change event was only coming through with the last typed character, i.e. the `e.target.value`s were

1. `e`
2. `x`
3. `p`
4. ...

I don't _think_ that would happen in a real browser environment, right?

File: src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx (hunk 1 above)
Comment (Proful) [reply src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx:37]: This feels more sensible.
`data-icon="asterisk"` more implementation details.

Comment (mpeyper) [reply src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx:37]: Just clarifying, do you mean what I have is the more sensible option?

Comment (Proful) [reply src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx:37]: yes I think so

**Pull request merged**